### PR TITLE
Address issue with updating with equality

### PR DESCRIPTION
**Pull request by LispEngineer**
See amalloy/ordered#31 for details.

Addressed by testing for identity instead of equality.

I am unfamiliar with the testing mechanism you used so I did not add a test for this case.

### CODE REVIEW

**Comment by jafingerhut** [comment]
Below is a suggested unit test you could add for your fix.  I have verified that it fails with the current code, but passes with your fix.
```
diff --git a/test/flatland/ordered/map_test.clj b/test/flatland/ordered/map_test.clj
index 65fbd39..87d4ff3 100644
--- a/test/flatland/ordered/map_test.clj
+++ b/test/flatland/ordered/map_test.clj
@@ -126,7 +126,10 @@
       (are [x] (= m x)
            (conj m nil)
            (merge m ())
-           (into m ())))))
+           (into m ()))))
+  (let [m (ordered-map :a '("spark") :b '("flare" "bolt"))]
+    (testing "assoc replaces values if not identical?"
+      (is (vector? (:a (assoc m :a (vec (:a m)))))))))
 
 (deftest object-features
   (let [m (ordered-map 'a 1 :b 2)]
```

File: project.clj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(defproject org.flatland/ordered "1.5.6-SNAPSHOT"
+(defproject org.flatland/ordered "1.5.6-DPF"
```
Comment (amalloy) [inline project.clj]: This version change doesn't belong in this PR.

**Comment by jafingerhut** [comment]
Created a separate PR for the new failing test case, modified as you suggested: https://github.com/amalloy/ordered/pull/33

**Comment by LispEngineer** [comment]
Requested changes submitted to my fork, both to version and added test. Please review and advise.

**Comment by amalloy** [comment]
Deployed to clojars as version 1.5.6. Thanks for the fix.

**Comment by LispEngineer** [comment]
Thanks for the quick turnaround, and to @jafingerhut for help with tests.
